drivers/boston_project_scorecard/app/models/boston_project_scorecard/financial_performance.rb

```diff
@@ -0,0 +1,116 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module BostonProjectScorecard
+  module FinancialPerformance
+    extend ActiveSupport::Concern
+    included do
+      def project_type_value
+        case project_type
+        when 3
+          'PSH'
+        when 9, 10
+          'PH'
+        when 13
+          'RRH'
+        else
+          'Other'
+        end
+      end
+
+      def project_type_score
+        return 6 if project_type == 3 # PSH
+        return unless project_type.in?(9, 10, 13)
+
+        3
+      end
+
+      def invoicing_options
+        {
+          'Invoices regularly submitted on time': 6,
+          'Usually on time, complete with few errors': 3,
+          'Submitted quarterly/not monthly': 1,
+        }
+      end
+
+      def invoicing_score
+        invoicing
+      end
+
+      def cost_efficiency_value
+        return unless actual_households_served&.positive?
+
+        (amount_agency_spent / actual_households_served.to_f).round(2)
+      end
+
+      def efficiency_score
+        return unless cost_efficiency_value.present?
+
+        return 6 if cost_efficiency_value <= 4000
+
+        0
+      end
+
+      def required_match_percent
+        return unless contracted_budget.present?
+
+        percentage(amount_agency_spent / contracted_budget.to_f)
+      end
+
+      def required_match_value
+        return unless required_match_percent.present?
+
+        percentage_string(required_match_percent)
+      end
+
+      def required_match_score
+        return unless required_match_percent.present?
+        return 6 if required_match_percent >= 25
+
+        0
+      end
+
+      def returned_funds_percent
+        return unless contracted_budget.present?
+
+        percentage(returned_funds / contracted_budget.to_f)
```

Comment:
if `contracted_budget` is 0, I think this will throw an error.  maybe `return unless contracted_budget.present? && contracted_budget.positive?`